File: Difficulty: Easy/Second Largest/second-largest.cpp
//{ Driver Code Starts
#include <bits/stdc++.h>

using namespace std;


// } Driver Code Ends
// User function template for C++
class Solution {
  public:
    // Function returns the second
    // largest elements
    int print2largest(vector<int> &arr) {
        // Code Here
        int largest = arr[0];
        int n=arr.size();
        
        for(int i=0; i<n; i++){
            if(arr[i]> largest){
                largest = arr[i];
            }
        }
        
        
        int secondl = -1;
        
        for(int j =0; j<n; j++){
            if(arr[j] == largest)
                continue;
            if(arr[j] > secondl){
                secondl = arr[j];
            }
            
        }

        return secondl;
        
        
        
    }
};

//{ Driver Code Starts.

int main() {
    int t;
    cin >> t;
    cin.ignore();
    while (t--) {
        vector<int> arr;
        string input;
        getline(cin, input);
        stringstream ss(input);
        int number;
        while (ss >> number) {
            arr.push_back(number);
        }
        Solution ob;
        int ans = ob.print2largest(arr);
        cout << ans << endl;
    }
    return 0;
}

// } Driver Code Ends

File: Difficulty: Easy/Second Largest/second-largest.java
//{ Driver Code Starts
// Initial Template for Java
import java.io.*;
import java.util.*;
import java.util.stream.Collectors;

public class Main {

    public static void main(String[] args) throws Exception {
        Scanner sc = new Scanner(System.in);
        int t = Integer.parseInt(sc.nextLine());
        while (t-- > 0) {
            String[] arr1Str = sc.nextLine().split(" ");
            int[] arr = Arrays.stream(arr1Str).mapToInt(Integer::parseInt).toArray();
            Solution ob = new Solution();
            int ans = ob.print2largest(arr);
            System.out.println(ans);
        }
    }
}

// } Driver Code Ends


// User function Template for Java

class Solution {
    public int print2largest(int[] arr) {
        // Code Here
        int max1=arr[0],max2=-1;
        for(int i=0;i<arr.length;i++){
            if(arr[i]>max1){
                max1=arr[i];
            }
        }
        for(int j=0;j<arr.length;j++){
            if(arr[j]>max2&&max1>arr[j]){
                max2=arr[j];
            }
        }
        return max2;
    }
}